vllm/model_executor/model_loader/weight_utils.py

```diff
@@ -654,6 +654,20 @@ def initialize_dummy_weights(
                 param.uniform_(low, high, generator=generator)
 
 
+def remap_modelopt_kv_scale_names(name: str):
+
+    modelopt_scale_name_dict = {
+        '.self_attn.k_proj.k_scale': '.self_attn.attn.k_scale',
+        '.self_attn.v_proj.v_scale': '.self_attn.attn.v_scale'
+    }
+
+    for scale_name in modelopt_scale_name_dict:
+        if name.endswith(scale_name):
+            remapped_name = name.replace(scale_name,
+                                         modelopt_scale_name_dict[scale_name])
+            return remapped_name
```

Comment:
Could you possibly put this logic into maybe_remap_kv_scale_name as a new case? I'm not sure why this needs to be separated out as a new pass in the weight utils